google/cloud/storage/tests/object_resumable_write_integration_test.cc

```diff
@@ -252,6 +252,116 @@ TEST_F(ObjectResumableWriteIntegrationTest, StreamingWriteSlow) {
   EXPECT_STATUS_OK(status);
 }
 
+TEST_F(ObjectResumableWriteIntegrationTest, WithXUploadContentLength) {
+  auto const MiB = 1024 * 1024L;
+  auto const chunk_size = 2 * MiB;
+
+  auto options = ClientOptions::CreateDefaultClientOptions();
+  ASSERT_STATUS_OK(options);
+  Client client(options->SetUploadBufferSize(chunk_size));
+
+  std::string bucket_name = flag_bucket_name;
+
+  auto const chunk = MakeRandomData(chunk_size);
+
+  for (auto const desired_size : {2 * MiB, 3 * MiB, 4 * MiB}) {
+    auto object_name = MakeRandomObjectName();
+    SCOPED_TRACE("Testing with desired_size = " + std::to_string(desired_size) +
+                 ", name=" + object_name);
+    auto os = client.WriteObject(
+        bucket_name, object_name, IfGenerationMatch(0),
+        CustomHeader("X-Upload-Content-Length", std::to_string(desired_size)));
+    auto offset = 0L;
+    while (offset < desired_size) {
+      auto const n = (std::min)(desired_size - offset, chunk_size);
+      os.write(chunk.data(), n);
+      ASSERT_FALSE(os.bad());
+      offset += n;
+    }
+
+    // This operation should fail because the object already exists.
```

Comment:
Q. Which part of the following indicates the failure?  Indeed, why should the random object exist?

Reply:
That was a cut&paste error, fixed here and elsewhere.